content/en/blog/2021/discovery-selectors/index.md

```diff
@@ -0,0 +1,85 @@
+
+## Background
+
+By default, Istio watches all namespaces/services/endpoints/pods information in a cluster. For example, in my Kubernetes cluster, I deployed the sleep service in the default namespace, and the httpbin service in the ns-x namespace. I’ve added sleep services to the mesh and I have no plan to add the httpbin service in the ns-x namespace to the mesh or have any service in the mesh interact with the httpbin service in the ns-x namespace.
+
+Use `istioctl pc endpoint` command to display all the endpoints for the sleep deployment:
+
+{{< image link="./endpoints-default.png" caption="Endpoints for Sleep Deployment" >}}
+
+Note that the httpbin service endpoint in the ns-x namespace is in the list of discovered endpoints. This may not be an issue when you only have a few services. However, when you have hundreds of services that don't interact with any of the services running in the Istio service mesh, you probably don't want your Istio control plane to watch these services and send their information to the sidecars of your services in the mesh.
+
+## Introduce DiscoverySelectors
+
+In Istio 1.10, we introduced the new `DiscoverySelectors` option to [MeshConfig](/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig), which is an array of Kubernetes [selectors](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements). The exact type will be `[]LabelSelector`, as defined [here](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements), allowing both simple selectors and set-based selectors. These selectors apply to labels on namespaces.
+
+You can configure each label selector for expressing a variety of use cases, including but not limited to:
+
+* Arbitrary label names/values, for example, all namespaces with label `istio-discovery=enabled`
+* A list of namespace labels using set-based selectors which carries OR semantics, for example, all namespaces with label `istio-discovery=enabled` OR `region=us-east1`
+* Inclusion and/or exclusion of namespaces, for example, all namespaces with label istio-discovery=enabled AND label key `app` equal to `helloworld`
+
+Note: `DiscoverySelectors` is not a security boundary. Istiod will continue to have access to all namespaces even when you have configured your `DiscoverySelectors`.
+
+## DiscoverySelectors in Action
+
+Assuming you know which namespaces to include as part of the service mesh, as a mesh administrator, you can configure `DiscoverySelectors` at installation time or post-installation by adding your desired discovery selectors to Istio’s MeshConfig resource. For example, you can configure Istio to discover only the namespace that has the label `istio-discovery=enabled`.
+
+1. Enable all namespaces with this label. Using our examples earlier, let’s enable namespace default with label `istio-discovery=enabled`.
```

Comment:
You need to indent the text bash to get the numbers correct. If you look at the actual page in the deploy test, you will see they are all 1.

Reply:
Thank YOU!!!